sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/aio/_azureappconfigurationproviderasync.py

```diff
@@ -0,0 +1,284 @@
+                try:
+                    j_object = json.loads(config.value)
+                    provider._dict[trimmed_key] = j_object
+                except json.JSONDecodeError:
+                    # If the value is not a valid JSON, treat it like regular string value
+                    provider._dict[trimmed_key] = config.value
+            else:
+                provider._dict[trimmed_key] = config.value
+    return provider
+
+def __buildprovider(connection_string:str, endpoint:str, credential,
+        key_vault_options:AzureAppConfigurationKeyVaultOptions):
+    provider = AzureAppConfigurationProvider()
+    headers = {}
+    correlation_context = "RequestType=Startup"
+
+    if key_vault_options and (
+        key_vault_options.credential or key_vault_options.secret_clients or key_vault_options.secret_resolver
+    ):
+        correlation_context += ",UsesKeyVault"
+
+    headers["Correlation-Context"] = correlation_context
+    useragent = USER_AGENT
+
+    if connection_string and endpoint:
+        raise AttributeError("Both connection_string and endpoint are set. Only one of these should be set.")
+
+    if connection_string:
+        #pylint:disable=protected-access
+        provider._client = AzureAppConfigurationClient.from_connection_string(
+            connection_string, user_agent=useragent, headers=headers
+        )
+        return provider
+    #pylint:disable=protected-access
+    provider._client = AzureAppConfigurationClient(endpoint, credential, user_agent=useragent, headers=headers)
+    return provider
+
+async def __resolve_keyvault_reference(config, key_vault_options:AzureAppConfigurationKeyVaultOptions, provider) -> str:
+    if key_vault_options is None:
+        raise AttributeError("Key Vault options must be set to resolve Key Vault references.")
+
+    if config.secret_id is None:
+        raise AttributeError("Key Vault reference must have a uri value.")
+
+    key_vault_identifier = KeyVaultSecretIdentifier(config.secret_id)
+
+    #pylint:disable=protected-access
+    referenced_client = provider._secret_clients.get(key_vault_identifier.vault_url, None)
+
+    if referenced_client is None and key_vault_options.credential is not None:
+        referenced_client = SecretClient(
+            vault_url=key_vault_identifier.vault_url, credential=key_vault_options.credential
+        )
+        provider._secret_clients[key_vault_identifier.vault_url] = referenced_client
+
+    if referenced_client:
+        return (
+            await referenced_client.get_secret(key_vault_identifier.name, version=key_vault_identifier.version)
+        ).value
+
+    if key_vault_options.secret_resolver is not None:
+        return key_vault_options.secret_resolver(config.secret_id)
+
+    raise AttributeError(
+        "No Secret Client found for Key Vault reference %s" % (key_vault_identifier.vault_url)
+    )
+
+def __is_json_content_type(content_type: str) -> bool:
```

Comment:
Is there any reason this couldn't simply be reused from the sync implementation?